runtime/src/main/java/org/corfudb/runtime/LogReplicationClient.java

```diff
@@ -0,0 +1,234 @@
+import org.corfudb.runtime.LogReplication.ClientInfoKey;
+import org.corfudb.runtime.LogReplication.SinksInfoVal;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.Table;
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class LogReplicationClient {
+
+    public static final String CORFU_SYSTEM_NAMESPACE = "CorfuSystem";
+    public static final String REGISTRATION_TABLE_NAME = "LogReplicationRegistrationTable";
+    public static final String LR_SOURCE_METADATA_TABLE_NAME = "LogReplicationSourceMetadataTable";
+
+    private static final int SYNC_THRESHOLD = 120;
+
+    private final CorfuStore corfuStore;
+
+    private final String clientName;
+    private final String modelEnum;
+
+    private final Logger log;
+
+    public LogReplicationClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        // TODO: auto assign model_enum based on the client name
+        this.modelEnum = "LOGICAL_GROUPS";
+
+        this.log = LoggerFactory.getLogger("lr-client");
+
+        corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, REGISTRATION_TABLE_NAME, StringKey.class,
+                StringKey.class, null, TableOptions.builder().build());
+    }
+
+    public boolean registerReplicationClient() {
+        try {
+            return IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+
+                    StringKey clientKey = StringKey.newBuilder()
+                            .setKey(clientName)
+                            .build();
+
+                    ClientInfo clientInfo = ClientInfo.newBuilder()
+                            .setClientName(clientName)
+                            .build();
+
+                    Table<StringKey, ClientInfo, Message> table = corfuStore.getTable(CORFU_SYSTEM_NAMESPACE, REGISTRATION_TABLE_NAME);
+
+                    if (txn.isExists(REGISTRATION_TABLE_NAME, clientKey)) {
+                        log.error("Client already registered");
+                        return false;
+                    }
+                    else {
+                        txn.putRecord(table, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+
+                    return txn.isExists(REGISTRATION_TABLE_NAME, clientKey);
```

Comment:
I guess this is redundant as it was already checked and if you reach this stage the tx was committed and it should exist.